fvm/environment/account_creator.go

```diff
@@ -36,6 +36,35 @@ type BootstrapAccountCreator interface {
 	)
 }
 
+type ParseRestrictedAccountCreator struct {
+	txnState *state.TransactionState
+	impl     AccountCreator
```

Comment:
will `ParseRestricted` check be applied to other operations too? if so it can be a generic decorator instead of just for AccountCreator. But it really depends on how it will be used.

Reply:
The ParsRestricted checks will be applied to everything, except Programs and ContractReader (and maybe Meter).

The different method signatures (in particular the arguments and return types) are interfering with using a generic decorator in general.  maybe missed something?

Reply:
o, i guess i can use closures for some of these, if the return type is only error

Reply:
yea i was thinking about closures. just curious to how code would look like with closures, if it is elegant enough, it may worth a try.